### DISCUSSION TITLE
Error in react-select 8 with react 16 and react-script 5

### DISCUSSION THREAD

**Discussion by sarang-damkondwar** [JedWatson/react-select#5849] [Q&A]
Hi,

I am not sure what is causing the issue, but it gives an error on react-select map function (screenshot attched).
react 16, react-scripts 5 and react-select 8
"react": "^16.14.0",
    "react-aux": "^1.1.0",
    "react-confirm-alert": "^2.0.5",
    "react-datepicker": "^1.8.0",
    "react-dom": "^16.4.1",
    "react-draggable": "^4.4.4",
    "react-fontawesome": "^1.6.1",
    "react-idle-timer": "^4.6.4",
    "react-portal": "^4.2.0",
    "react-pro-sidebar": "^0.4.4",
    "react-redux": "^5.0.7",
    "react-router": "^4.3.1",
    "react-router-dom": "5.3.3",
    "react-scripts": "^5.0.1",
    "react-select": "^5.7.7",

<img width="321" alt="image" src="https://github.com/JedWatson/react-select/assets/88040971/1743ee9f-7ab6-476f-bb0b-0923ee6eccf7">

Thanks, in advance.

**Comment by manjushsh** [comment]
@sarang-damkondwar , Can you share a code example with bug replicated? If possible, on codesandbox

**Reply by sarang-damkondwar** [reply]
Unable to share code but I can try explaining here,
It seems this the error occurs when 'options' variable becomes null. before it was working fine.
 
Note: I have added custom filters on options and while it filters options variable goes null, which impacts and sets it (options variable) to null.

**Reply by manjushsh** [reply]
Options should be Array right? can you try passing empty array if no values instead of null?

**Comment by Azharhayat271** [comment]
this error occurs when you are trying to fetch the data from API but API is returning an object or null value other than array then there are chance of this error for more detail kindly share the code.
